core/saltfilter/filter.go

```diff
@@ -0,0 +1,67 @@
+// Package saltfilter used for reuse detection ONLY
+package saltfilter
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// Those suggest value are all set according to
+// https://github.com/shadowsocks/shadowsocks-org/issues/44#issuecomment-281021054
+const (
+	SuggestEntries            = int(1e6)
+	SuggestFalsePositiveRate  = 1e-6
+	defaultBloomRingSlotCount = 5
+)
+
+// Support override ability while in compiling via `-ldflags -X`
+var (
+	entries           string
+	falsePositiveRate string
+)
+
+var singleton *BloomRing
```

Comment:
Singleton is really a foreign concept in Go… Also it kinda weird to have this package-level global variable when it's better to explicitly use a BloomRing.